cmd/clusterctl/cmd/config_repositories.go

```diff
@@ -71,13 +116,33 @@ var configRepositoryCmd = &cobra.Command{
 
 func init() {
 	configRepositoryCmd.Flags().StringVarP(&cro.output, "output", "o", RepositoriesOutputText,
-		fmt.Sprintf("Output format. Valid values: %v.", RepositoriesOutputs))
+		fmt.Sprintf("Output format. Valid values: %v.", configRepositoriesGetSupportedOutputs()))
+	configRepositoryCmd.Flags().StringVar(&cro.provider, "provider", "",
```

Comment:
`--output name` and `--provider` flags of `clusterctl config repositories` command are used for completion for `--bootstrap`, `--core`, `--control-plane`, and `--infrastructure` flags of `clusterctl config provider` command.

- https://github.com/kubernetes-sigs/cluster-api/pull/3808/files#diff-2bfa2cc0fd5aa1ab704fc6fe8817d76dafbc142abc958d687df7b6c0d6f49c2bR126-R132

Reply:
I'm just trying to understand in order to provide more valuable feedback, and one or more issue explaining what we want to achieve could really help me in this.

For example, now I'm struggling with questions like:
Why it is required to add a --provider flag for completion? or, why it is required to add another format to `--output` for completion? I was expecting completion to document whatever my command is, not to alter my commands...
Also, in case assuming we should alter commands for completion, why it is required to alter `clusterctl config repositories` only and not the other commands?

Reply:
> Why it is required to add a --provider flag for completion? or, why it is required to add another format to --output for completion?

For example, the value of the `--bootstrap` option is the name of the bootstrap provider. To complete this, we need to get a list of bootstrap provider names from somewhere in the following format:

```
aws-eks
kubeadm
talos
```

To get a list of providers with the `clusterctl` command, there is `clusterctl config repositories` command. However, it cannot output only names. Also, it cannot filter by provider type.

Therefore, I add the `-o name` option to output only names and the `--provider` option to filter by provider type to `clusterctl config repositories` command.

If you do `clusterctl init --bootstrap <tab>`, `clusterctl config repositories -o name --provider bootstrap` will be executed behind the scenes, and the execution result will be displayed as completion candidates.

```
$ clusterctl init --bootstrap <tab>
aws-eks  kubeadm  talos
```

Other `--core`,` --control-plane`, ʻinfrastructure` work in the same way.